CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -272,7 +269,7 @@ public function buildQuickForm() {
    */
   public static function formRule($fields, $files, $self) {
     $errors = [];
-    if ($self->_paymentType == 'owed' && $fields['total_amount'] > $self->_owed) {
+    if ($self->_paymentType == 'owed' && (int) $fields['total_amount'] > (int) $self->_owed) {
```

Comment:
According to https://lab.civicrm.org/dev/core/issues/1463#note_28671, maybe, the complete check should be removed? Hmm, but that might add some more discussion as what needs to be done to the main contribution amount when the form is submitted with > owed amount?